### PR TITLE
SDIT-1367 Add create and get incident mappings

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncidentMappingResource.kt [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        Content(
+          mediaType = "application/json",
+          schema = Schema(implementation = IncidentMappingDto::class),
+        ),
+      ],
+    ),
+    responses = [
+      ApiResponse(responseCode = "201", description = "Mapping entry created"),
+      ApiResponse(
+        responseCode = "409",
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncidentMappingResource.kt:48]: I think this is duplicated below?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncidentMappingResourceIntTest.kt [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+      val duplicateIncident = responseBody.moreInfo?.duplicate!!
+      with(duplicateIncident) {
+        assertThat(incidentId).isEqualTo("4444")
+        assertThat(nomisIncidentId).isEqualTo(21)
+        assertThat(mappingType).isEqualTo("NOMIS_CREATED")
+      }
+    }
+
+    @Test
+    internal fun `create mapping does not error when the same mapping already exists for the same adjustment`() {
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncidentMappingResourceIntTest.kt:152]: copy n paste typo "adjustment"

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncidentMappingResource.kt (hunk 1 above)
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncidentMappingResource.kt:48]: will check and remove - think I was a bit copy and paste happy

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncidentMappingResourceIntTest.kt (hunk 2 above)
Comment (karenmillermoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncidentMappingResourceIntTest.kt:152]: will fix in next PR - when that eventually comes along

**Pull request merged**